birdhouse/components/geoserver/pre-docker-compose-up

```diff
@@ -1,10 +1,14 @@
 #!/bin/sh
 
-THIS_FILE="`realpath "$0"`"
-THIS_DIR="`dirname "$THIS_FILE"`"
+THIS_FILE="$(readlink -f "$0" || realpath "$0")"
+THIS_DIR="$(dirname "$THIS_FILE")"
 COMPOSE_DIR="$THIS_DIR/../.."
 
+if [ -f "${COMPOSE_DIR}/scripts/logging.include.sh" ]; then
+  . "${COMPOSE_DIR}/scripts/logging.include.sh"
```

Comment:
Oh I just spotted an error with this script !  It did not source `read-configs.include.sh` and did not call `read_configs`.  This is bad because it uses `GEOSERVER_DATA_DIR` which is a delayed eval var !

Reply:
Looking at this file made me realize something else as well.
`COMPOSE_DIR` is explicitly set from the `THIS_FILE`/`THIS_DIR` variables, which themselves assume the file is called directly instead of sourced. Is this still valid, considering a previous comment where it was mentioned that `COMPOSE_DIR` could be provided explicitly to find the utilities from any entry point?

Reply:
> Looking at this file made me realize something else as well.
> `COMPOSE_DIR` is explicitly set from the `THIS_FILE`/`THIS_DIR` variables, which themselves assume the file is called directly instead of sourced. Is this still valid, considering a previous comment where it was mentioned that `COMPOSE_DIR` could be provided explicitly to find the utilities from any entry point?

@fmigneault 

`pre-docker-compose-up` is an actual script, it is not being sourced by another script.  So `pre-docker-compose-up` can set `COMPOSE_DIR` via its own `THIS_FILE`/`THIS_DIR` variables.

`read-configs.include.sh` is being sourced by other scripts.  So the `THIS_FILE`/`THIS_DIR` variables are values from the other scripts, which will be different for each other script.  Therefore `read-configs.include.sh` can not reliably derive `COMPOSE_DIR` via `THIS_FILE`/`THIS_DIR` variables because `read-configs.include.sh` does not own those variables.

Hope it's clearer?